File: benchmarks/values/bool_sieve.go
package main

import (
	"fmt"
	"time"
)

func main() {
	t0 := time.Now()

	n := 1000000
	isPrime := make([]bool, n+1)
	for i := range isPrime {
		isPrime[i] = true
	}
	isPrime[0] = false
	isPrime[1] = false

	i := 2
	for i*i <= n {
		if isPrime[i] {
			j := i * i
			for j <= n {
				isPrime[j] = false
				j += i
			}
		}
		i++
	}

	count := 0
	for k := 0; k <= n; k++ {
		if isPrime[k] {
			count++
		}
	}

	elapsed := time.Since(t0)
	fmt.Println(count)
	fmt.Printf("elapsed: %.3fs\n", elapsed.Seconds())
}

File: benchmarks/values/bool_sieve.py
import time

t0 = time.perf_counter()

n = 1_000_000
is_prime = [True] * (n + 1)
is_prime[0] = False
is_prime[1] = False

i = 2
while i * i <= n:
    if is_prime[i]:
        j = i * i
        while j <= n:
            is_prime[j] = False
            j += i
    i += 1

count = 0
for k in range(n + 1):
    if is_prime[k]:
        count += 1

t1 = time.perf_counter()
print(count)
print(f"elapsed: {t1 - t0:.3f}s")
